x-pack/plugins/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -67,11 +68,11 @@ async function fetchAgentPolicy(soClient: SavedObjectsClientContract, id: string
 export async function getFullAgentPolicy(
   soClient: SavedObjectsClientContract,
   id: string,
-  options?: { standalone: boolean }
+  options?: { standalone?: boolean; agentPolicy?: AgentPolicy }
 ): Promise<FullAgentPolicy | null> {
   const standalone = options?.standalone ?? false;
 
-  const agentPolicy = await fetchAgentPolicy(soClient, id);
+  const agentPolicy = options?.agentPolicy ?? (await fetchAgentPolicy(soClient, id));
```

Comment:
maybe worth checking `package_policies` is defined in `agentPolicy` and if not fetching the agent policy, as we often fetch the agent policy without the related package policy to avoid loading too much objects

Reply:
done !!